gno.land/pkg/sdk/vm/builtins.go

```diff
@@ -41,7 +41,19 @@ func (bnk *SDKBanker) SendCoins(b32from, b32to crypto.Bech32Address, amt std.Coi
 }
 
 func (bnk *SDKBanker) TotalCoin(denom string) int64 {
-	panic("not yet implemented")
+	if denom == "" {
+		panic("empty denom")
+	}
+	accounts := bnk.vmk.acck.GetAllAccounts(bnk.ctx)
+	var totalAmount int64
+
+	for _, acc := range accounts {
+		if acc == nil {
+			continue
```

Comment:
not scalable.

Reply:
I modified this function to process in a single loop. This is what I could think of for now. What do you think?

https://github.com/gnolang/gno/pull/3899/commits/0d751b65e67e5e45fd70e70af3da6d3fd9a48e80

Reply:
This needs a change in the underlying banker module, so that it keeps track of the available tokens, rather than iterating on the accounts here.